tiledb/sm/filesystem/posix.cc

```diff
@@ -514,6 +514,17 @@ Status Posix::sync(const std::string& path) {
 
 Status Posix::write(
     const std::string& path, const void* buffer, uint64_t buffer_size) {
+  // Check for valid inputs before attempting the actual
+  // write system call. This is to avoid a bug on macOS
+  // Ventura 13.0 on Apple's M1 processors.
+  if (buffer == nullptr) {
+    return LOG_STATUS(Status_IOError("data buffer must not be nullptr"));
```

Comment:
This should probably throw `invalid_argument` and not return.

Reply:
Done.